build/helper/metadata_filters.py

```diff
@@ -6,9 +6,9 @@ def filter_codegen_functions(functions):
     return {k: v for k, v in functions.items() if v['codegen_method'] != 'no'}
 
 
-def filter_input_parameters(parameters, session_name='vi'):
+def filter_input_parameters(parameters):
     '''Returns list of parameters that includes only input parameters, except the session parameter if it exists'''
```

Comment:
repeated capability as well...
But... how about getting rid of this function altogether and use the all-powerful filter function?

Reply:
Which all powerful filter function? The one added when adding the 'len' mechanism? That one only looks for a specific mechanism.

Reply:
filter_parameters() in build/helper/codegen_helper.py

Reply:
Do you want me to make that change in this PR? There would be other filter functions that should be removed as well. Do I get rid of all of them?

Reply:
> Do you want me to make that change in this PR?

Up to you.

> There would be other filter functions that should be removed as well. Do I get rid of all of them?

The ones for filtering parameters, yeah, we can get rid of most if not all.